pkg/controller/jobframework/reconciler.go

```diff
@@ -600,6 +600,33 @@ func (r *JobReconciler) handleJobWithNoWorkload(ctx context.Context, job Generic
 	if err != nil {
 		return err
 	}
+
+	// Create PodTemplate object
+	if jwpt, implements := job.(JobWithPodTemplate); implements {
```

Comment:
This shouldn't be optional. You can still create a PodTemplate object from the returned job.PodSets()

Reply:
removed as optional